src/MedTrackerAPI/Features/Supplies/AddSupply.cs

```diff
@@ -0,0 +1,65 @@
+using FluentValidation;
+using MediatR;
+using MedTrackerAPI.Endpoints;
+using MedTrackerAPI.Infrastructure;
+
+namespace MedTrackerAPI.Features.Supplies;
+
+public static class AddSupply
+{
+    public record Response (int SupplyId, string? Description, string? Manufacturer, string? PartNumber, string? LotNumber, int DeviceId);
+    
+    public class AddSupplyCommandValidator : AbstractValidator<AddSupplyCommand>
+    {
+        public AddSupplyCommandValidator()
+        {
+            RuleFor(x => x.Description).NotEmpty().WithMessage("Description is required.");
+            RuleFor(x => x.PartNumber).NotEmpty().WithMessage("Part number is required.");
+            RuleFor(x => x.LotNumber).NotEmpty().WithMessage("Lot number is required.");
+            RuleFor(x => x.DeviceId).NotEmpty().WithMessage("Device Id is required.");
+        }
+    }
+
+    public sealed class Endpoint : IEndpoint
+    {
+        public void MapEndpoint(IEndpointRouteBuilder app)
+        {
+            app.MapPost("supplies", async (IMediator mediator, AddSupplyCommand command) =>
+            {
+                var supply = await mediator.Send(command);
+
+                return Results.Created("", supply);
+            }).WithTags("Supplies")
+            .AddEndpointFilter<ValidationFilter<AddSupplyCommand>>();
+        }
+    }
+    
+    public class AddSupplyCommand : IRequest<Response>
+    {
+        public required string Description { get; set; }
```

Comment:
Should I remove the required here even though it is required in the DB and the Validation will force it? That would require me to make it a nullable field on the Command here. I didn't include the required on the CreateDeviceCommand, but unsure why I wouldn't if it's required on the backend anyway.

Reply:
I think the required might return a 400 request from the framework and not your validation, so a nullable might be necessary. Otherwise I’d do required I think

Reply:
I tested with both required and removing it. I couldn't tell the difference if one was coming from the Framework or the validator. They return 400's looked identical. I removed the required keyword though to match the CreateDevice. I may change and add the required back later though after some research. 

Created issue #19 to come back to.